jdaviz/configs/imviz/plugins/parsers.py

```diff
@@ -371,37 +371,17 @@ def _nddata_to_glue_data(ndd, data_label):
     if ndd.data.ndim != 2:
         raise ValueError(f'Imviz cannot load this NDData with ndim={ndd.data.ndim}')
 
-    for attrib, sub_attrib in zip(
-            ['data', 'mask', 'uncertainty'],
-            [None, None, 'array']
-    ):
+    for attrib in ('data', 'mask', 'uncertainty'):
         arr = getattr(ndd, attrib)
         if arr is None:
             continue
-        cur_data = Data()
+        comp_label = attrib.upper()
+        cur_label = f'{data_label}[{comp_label}]'
+        cur_data = Data(label=cur_label)
         cur_data.meta.update(standardize_metadata(ndd.meta))
         if ndd.wcs is not None:
             cur_data.coords = ndd.wcs
         raw_arr = arr
-
-        if sub_attrib is not None:
-            # since NDDataArray.uncertainty may be an object like
-            # StdDevUncertainty, we need to take another attr
-            # like StdDevUncertainty.array:
```

Comment:
I had intentionally double-checked each new NDData to ensure they were meant to be WCS only by checking for the `_WCS_ONLY` key _and_ that the data are all NaN.  But I agree that this revision isn't strictly necessary, not least because I wrote the translator for the `StdDevUncertainty` 😉.

Reply:
Would there ever be a case where user would use `_WCS_ONLY` with real data for some other reason? Otherwise, I don't see why we need one extra check, but maybe I am missing something here.

Reply:
> Would there ever be a case where user would use _WCS_ONLY with real data for some other reason?

Probably not, which is why I have no objections.